mplex/src/lib.rs

```diff
@@ -107,77 +111,120 @@ impl<C> Clone for Multiplex<C> {
 
 // Struct shared throughout the implementation.
 struct MultiplexInner<C> {
+    // Errored that happend earlier. Should poison any attempt to use this `MultiplexError`.
+    error: Result<(), IoError>,
     // Underlying stream.
-    inner: Fuse<Framed<C, codec::Codec>>,
+    inner: executor::Spawn<Fuse<Framed<C, codec::Codec>>>,
```

Comment:
Could you explain a bit what made this change and the introduction of `Notifier` necessary?

Reply:
To be honest I thought it was necessary for multiplexing to work correctly, then realized it was not.